tests/testthat/test-prophet.R

```diff
@@ -0,0 +1,663 @@
+
+  options <- analysisOptions("Prophet")
+  options$dependent <- "contNormal"
+  options$time <- "debString"
+  options$changepoints <- "contBinom"
+  options$mcmcSamples <- 10
+  options$predictionSavePath <- ""
+  set.seed(1)
+  results <- runAnalysis("Prophet", "prophetTest.csv", options)
+  expect_identical(results[["status"]], "validationError", label = "'Changepoints' must be a logical variable (e.g., 0/1)")
+
+  options$changepoints <- ""
+  options$historyIndicator <- "contBinom"
+  set.seed(1)
+  results <- runAnalysis("Prophet", "prophetTest.csv", options)
+  expect_identical(results[["status"]], "validationError", label = "'History Indicator' must be a logical variable (e.g., 0/1)")
+
+  options$historyIndicator <- ""
+  options$capacity <- "contcor1"
+  options$minimum <- "contNarrow"
+  options$historyIndicator <- "histIdx"
+  options$growth <- "logistic"
+  set.seed(1)
+  results <- runAnalysis("Prophet", "prophetTest.csv", options)
+  expect_identical(results[["status"]], "validationError", label = "'Carrying Capacity' must always be larger than 'Saturating Minimum'")
+
+  options$minimum <- ""
+  options$constantMinimum <- 0
+  set.seed(1)
+  results <- runAnalysis("Prophet", "prophetTest.csv", options)
+  expect_identical(results[["status"]], "validationError", label = "'Carrying Capacity' must always be larger than 'Constant Saturating Minimum'")
+
+  options$capacity <- ""
+  options$constantCapacity <- -1
+  options$minimum <- "contNarrow"
+  set.seed(1)
+  results <- runAnalysis("Prophet", "prophetTest.csv", options)
+  expect_identical(results[["status"]], "validationError", label = "'Constant Carrying Capacity' must always be larger than 'Saturating Minimum'")
+
+  options$minimum <- ""
+  options$historyIndicator <- ""
+  options$constantMinimum <- 0
+  set.seed(1)
+  results <- runAnalysis("Prophet", "prophetTest.csv", options)
+  expect_identical(results[["status"]], "validationError", label = "'Constant Carrying Capacity' must always be larger than 'Constant Saturating Minimum'")
+
+  options <- analysisOptions("Prophet")
+  options$dependent <- "contNormal"
+  options$time <- "dateDay"
+  options$mcmcSamples <- 10
+  options$predictionSavePath <- ""
+  options$seasonalities <- list(list(name = "---",
+                                     period = 4,
+                                     unit = dateTimeUnits[i],
+                                     priorSigma = 10,
+                                     fourierOrder = 3,
+                                     mode = "additive"))
+  set.seed(1)
+  results <- runAnalysis("Prophet", "prophetTest.csv", options)
+  expect_identical(results[["status"]], "validationError", label = "Seasonality names must only contain letters, number, dots, or underscores and must start with letters or dots that are not followed by a number")
```

Comment:
The test here fails because of `dateTimeUnites[i]` - variable `i` does not exist. Not sure what you want to do with it, but it needs a fix ;)